balance_strategy.go

```diff
@@ -127,3 +155,926 @@ func balanceStrategyHashValue(vv ...string) uint32 {
+	for _, assignment := range assignments {
+		if assignment == topic {
+			return true
+		}
+	}
+	return false
+}
+
+func sortPartitions(currentAssignment map[string][]topicPartitionAssignment, partitionsWithADifferentPreviousAssignment map[topicPartitionAssignment]consumerGenerationPair, isFreshAssignment bool, partition2AllPotentialConsumers map[topicPartitionAssignment][]string, consumer2AllPotentialPartitions map[string][]topicPartitionAssignment) []topicPartitionAssignment {
+	sortedPartitions := make([]topicPartitionAssignment, 0)
+	if !isFreshAssignment && areSubscriptionsIdentical(partition2AllPotentialConsumers, consumer2AllPotentialPartitions) {
+		// if this is a reassignment and the subscriptions are identical (all consumers can consumer from all topics)
+		// then we just need to simply list partitions in a round robin fashion (from consumers with
+		// most assigned partitions to those with least)
+		assignments := filterAssignedPartitions(currentAssignment, partition2AllPotentialConsumers)
+
+		// use priority-queue to evaluate consumer group members in descending-order based on
+		// the number of topic partition assignments (i.e. consumers with most assignments first)
+		pq := make(assignmentPriorityQueue, len(assignments))
+		i := 0
+		for consumerID, consumerAssignments := range assignments {
+			pq[i] = &consumerGroupMember{
+				id:          consumerID,
+				assignments: consumerAssignments,
+				index:       i,
+			}
+			i++
+		}
+		heap.Init(&pq)
+
+		for {
+			// loop until no consumer-group members remain
+			if pq.Len() == 0 {
+				break
+			}
+			member := pq[0]
+
+			// partitions that were assigned to a different consumer last time
+			prevPartitions := make([]topicPartitionAssignment, 0)
+			for partition := range partitionsWithADifferentPreviousAssignment {
+				// from partitions that had a different consumer before, keep only those that are assigned to this consumer now
+				if memberAssignmentsIncludeTopicPartition(member.assignments, partition) {
+					prevPartitions = append(prevPartitions, partition)
+				}
+			}
+
+			if len(prevPartitions) > 0 {
+				// if there is a partition on this consumer that was assigned to another consumer before mark it as good options for reassignment
+				partition := prevPartitions[0]
+				prevPartitions = append(prevPartitions[:0], prevPartitions[1:]...)
+				member.assignments = removeTopicPartitionFromMemberAssignments(member.assignments, partition)
+				sortedPartitions = append(sortedPartitions, partition)
+				heap.Fix(&pq, member.index)
+			} else if len(member.assignments) > 0 {
+				// otherwise, mark any other one of the current partitions as a reassignment candidate
+				partition := member.assignments[0]
+				member.assignments = append(member.assignments[:0], member.assignments[1:]...)
+				sortedPartitions = append(sortedPartitions, partition)
+				heap.Fix(&pq, member.index)
```

Comment:
No need to keep member index. The first item in the slice is the largest, which is how you declare `member` above to begin with. So, we know the index is zero here (and just above). Essentially, all uses of member.index can be replaced with 0 and the index field can be removed entirely.